order-book/src/storage/order.rs

```diff
@@ -0,0 +1,113 @@
+use crate::storage::common_storage::Percent;
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+pub type OrderId = u64;
+pub type Timestamp = u64;
+
+pub const MINUTE_IN_SECONDS: Timestamp = 60;
+pub const HOUR_IN_SECONDS: Timestamp = 60 * MINUTE_IN_SECONDS;
+pub const DAY_IN_SECONDS: Timestamp = 24 * HOUR_IN_SECONDS;
+
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode, PartialEq)]
+pub enum OrderStatus {
+    Pending,
+    PartiallyFilled,
+}
+
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode)]
+pub enum OrderDuration {
+    Minutes(u8),
+    Hours(u8),
+    Days(u8),
+}
+
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode)]
+pub struct Order<M: ManagedTypeApi> {
```

Comment:
Shouldn't we have an additional state here? As per the tech specs, I think we should have an order status logic with the following options: Pending, PartiallyFilled, Executed, Cancelled, Expired.